From: Olivier Fourdan <ofourdan@redhat.com>
Date: Thu, 20 Jan 2022 15:36:36 +0100
Subject: gdm.rules: Prefer Wayland with NVIDIA >= 510

NVIDIA driver version 510 and above have support for GBM, use Wayland by
default with NVIDIA proprietary driver version 510 and above.

For versions between 470 and 510, prefer Xorg as before.
---
 data/61-gdm.rules.in | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/data/61-gdm.rules.in b/data/61-gdm.rules.in
index bbc6a00..e9c7269 100644
--- a/data/61-gdm.rules.in
+++ b/data/61-gdm.rules.in
@@ -10,7 +10,9 @@ IMPORT{cmdline}="nomodeset", GOTO="gdm_disable_wayland"
 LABEL="gdm_nomodeset_end"
 
 # Disable wayland when nvidia modeset is disabled or when drivers are a lower
-# version than 470, in any case always prefer Xorg
+# version than 470,
+# For versions above 470 but lower than 510 prefer Xorg,
+# Above 510, prefer Wayland.
 KERNEL!="nvidia_drm", GOTO="gdm_nvidia_drm_end"
 SUBSYSTEM!="module", GOTO="gdm_nvidia_drm_end"
 ACTION!="add", GOTO="gdm_nvidia_drm_end"
@@ -18,6 +20,9 @@ ACTION!="add", GOTO="gdm_nvidia_drm_end"
 ATTR{parameters/modeset}!="Y", GOTO="gdm_disable_wayland"
 # disable wayland for nvidia drivers versions lower than 470
 ATTR{version}=="4[0-6][0-9].*|[0-3][0-9][0-9].*|[0-9][0-9].*|[0-9].*", GOTO="gdm_disable_wayland"
+# For nvidia drivers versions Above 510, keep Wayland by default
+ATTR{version}=="[5-9][1-9][0-9].*", GOTO="gdm_end"
+# For nvidia drivers versions 470-495, prefer Xorg by default
 GOTO="gdm_prefer_xorg"
 LABEL="gdm_nvidia_drm_end"
 
